minichain/mempool.py

```
@@ -17,62 +16,62 @@ def add_transaction(self, tx):
         with self._lock:
-            for tx in transactions:
-                pool = self._pool.get(tx.sender)
-                if pool and tx.nonce in pool:
-                    del pool[tx.nonce]
-                    self._size -= 1
-                    if not pool:
-                        del self._pool[tx.sender]
+            keys_to_remove = {(tx.sender, tx.nonce) for tx in transactions}
+            self._list = [tx for tx in self._list if (tx.sender, tx.nonce) not in keys_to_remove]
 
     def __len__(self):
```

Ruff (0.15.15)

[warning] Missing return type annotation for special method `__len__`

Add return type annotation: `int`

(ANN204)